app/services/designation_and_undesignation/same_designation.rb

```diff
@@ -0,0 +1,51 @@
+module DesignationAndUndesignation
```

Comment:
This is really good. Can you indicate with methods are part of the public API by putting them at the top of the class (underneath `initialize`) and then using `private` or `protected` (for base class). Same for all subclasses. Then it will be clear what methods we can/can't use in the rest of the app.